### ISSUE TITLE
unmeow writes all zeros to output file

### ISSUE THREAD

**Issue by JnyJny** [JnyJny/meowmeow#3]
The ```unmeow``` command creates files full of NULLs, e.g.:

```bash
$ meow < Makefile | unmeow | cat -v
^@^@^@^@^@^@^@^@^@^@^@^@...
```

**Comment by JnyJny** [comment]
In the  ```mm_decode()``` function, the bit values are decoded using ```isupper(3)```:

```C
byte.field.f0 = isupper(buf[0]);
```

The assumption made here is that isupper returns a boolean value when in fact it returns an ```int``` and the values returned are zero and non-zero.  On macOS 10.11.x, ```isupper``` returns zero and one and the code above works as intended.

Compiling on Redhat Linux, ```unmeow``` generates files composed solely of zeros. The values returned by ```isupper(3)``` on RedHat Linux are zero and 0x00000100.   When 0x0100 is assigned to a 1-bit wide bitfield, the high bit (a zero) of the int is taken for the value of the bitfield. And thus all NULLs written to the file. 

The fix for this problem is relatively simple.  Since zero is the only guaranteed sentinel value, use the following assignment construction instead:

```C
byte.field.f0 = isupper(buf[0])==0?0:1;
```

The return value of ```isupper(3)``` is tested against zero in the conditional assignment and the values we expected are supplied as literal constants.

**Comment by JnyJny** [comment]
Fixed with: 27f14db8fc2abe6e8522fb2f0c575069f73b6c54
